### DISCUSSION TITLE
Multiple carousels in one Page

### DISCUSSION THREAD

**Discussion by AhmedAbdulrahman** [davidjerleke/embla-carousel#282] [Help] [answered]
Hey there! 

I'm trying to rung multiple carousels in a single page and only first active and other 2, 3, or more hidden. It has to get `reinit` when `tab` button is clicked. I'm a little stuck trying to solve this. I can't seem to get the `embla` carousel to `reinit` on tab change. In this kind of situation, do I need to add multiple `carousels` or just one but hide `tab` contents when button is clicked?

Any guidance here would be appreciated!

Example:
https://codesandbox.io/s/embla-carousel-init-slider-onclick-ep12yu

**Comment by davidjerleke** [comment, marked as answer]
Hi @AhmedAbdulrahman,

Thanks for your question. Two things that caught my attention:

- The Embla version you're using is **highly** outdated, like `2.x.x` something. I don't even think the `embla.reInit()` method was introduced back then. I've updated the version to the latest which is `6.2.0`.
- You need a way to point to the correct carousel instance when activating a tab on click. I created a carousel map where you can access a carousel instance by using the `id` attribute on the `tab-content` nodes. Using this map, it's easy to **re-initialise** the carousel that becomes visible.

I've modified your [CodeSandbox here](https://codesandbox.io/s/embla-carousel-init-slider-onclick-forked-lpoul2). And here's the [app view](https://lpoul2.csb.app/) for the same sandbox.

Let me know how it goes.

Best,
David

**Reply by davidjerleke** [reply]
@AhmedAbdulrahman please mark [my comment](https://github.com/davidcetinkaya/embla-carousel/discussions/282#discussioncomment-2196432) as answer if it answers your question.

Best,
David

**Reply by AhmedAbdulrahman** [reply]
Thanks a lot sorry for late response I was a way, It really helped a lot and really appreciate your notes :)

**Comment by AhmedAbdulrahman** [comment]
Btw, I tried this scale example of yours [CodeSandBox](https://codesandbox.io/s/embla-carousel-scale-vanilla-forked-vjf37f) do we need to call the scale on `reInit`?  @davidcetinkaya 

when first tab is active 
![image](https://user-images.githubusercontent.com/10485129/154906977-e49aafc8-9892-43d2-9c53-cb45845c9e01.png)

when second / third tab is active (reset scale)
![image](https://user-images.githubusercontent.com/10485129/154907218-231d6d7b-1e4b-4866-b8ab-bce6460ad6fd.png)

but it works when I start navigation on left and right
![image](https://user-images.githubusercontent.com/10485129/154907132-8dde07d5-0735-4d07-b803-9b4972d86a12.png)

**Reply by davidjerleke** [reply]
@AhmedAbdulrahman yes, you need to do that because the scale effect needs to be re-calculated then. You can do something along these lines:

```js
function yourScaleFunction() {
  // scale magic function
}

embla
  .on('init', yourScaleFunction)
  .on('reInit', yourScaleFunction)
```

Cheers,
David
